test/hybrid2d.jl

```diff
@@ -48,77 +48,83 @@ function hvspace_2D(
+# V_face = Covariant3Vector
+# V_center = Covariant12Vector
+# V = V_center + V_face
 
+# divergence(V) = horz_div(V) + vert_div(V)
+# divergence(V) = horz_div(V_center) + horz_div(V_face) + vert_div(V_center) + vert_div(V_face)
+
+
+# 1) horz_div(V_center): project to Contravariant1 + Contravariant2, take spectral derivative
+# 2) horz_div(V_face): project to Contravariant1 + Contravariant2, interpolate to center, take spectral derivative
+#   - will be zero if orthogional geom
+# 3) vert_div(V_center): project to Contravariant3, interpolate to face, take FD deriv
+#   - will be zero if orthogional geom
+# 4) vert_div(V_face): project to Contravariant3, take FD deriv
+
+
+
+
+
+@testset "1D SE, 1D FD Extruded Domain vertical advection operator" begin
+
+    # Advection Operator
+    # c ∂_z f
+    # for this test, we use f(z) = sin(z) and c = 1
+    # => c ∂_z f = cos(z)
     hv_center_space, hv_face_space = hvspace_2D()
-    V =
-        Geometry.Cartesian13Vector.(
-            zeros(Float64, hv_face_space),
-            ones(Float64, hv_face_space),
-        )
 
-    function rhs!(dudt, u, _, t)
+    function rhs!(f)
+        adv = zeros(eltype(f), hv_center_space)
         A = Operators.AdvectionC2C(
-            bottom = Operators.SetValue(sin(-t)),
+            bottom = Operators.SetValue(0.0),
             top = Operators.Extrapolate(),
         )
-        return @. dudt = -A(V, u)
+        return @. adv = A(c, f)
     end
 
-    U = sin.(Fields.coordinate_field(hv_center_space).z)
-    dudt = zeros(eltype(U), hv_center_space)
-    rhs!(dudt, U, nothing, 0.0)
+    # advective velocity
+    c = Geometry.Cartesian3Vector.(ones(Float64, hv_face_space),)
+    # scalar-valued field to be advected
+    f = sin.(Fields.coordinate_field(hv_center_space).z)
+    # evaluate tendency
+    adv = rhs!(f)
 
-    using OrdinaryDiffEq
-    Δt = 0.01
-    prob = ODEProblem(rhs!, U, (0.0, 2π))
-    sol = solve(prob, SSPRK33(), dt = Δt)
-
-    htopo = Spaces.topology(hv_center_space)
-    for h in 1:Topologies.nlocalelems(htopo)
-        sol_column_field = ClimaCore.column(sol.u[end], 1, 1, h)
-        ref_column_field = ClimaCore.column(U, 1, 1, h)
-        @test norm(sol_column_field .- ref_column_field) ≤ 5e-2
-    end
+    @test norm(adv .- cos.(Fields.coordinate_field(hv_center_space).z)) ≤ 5e-2
 end
 
-@testset "1D SE, 1D FV Extruded Domain ∇ ODE Solve horzizontal" begin
+@testset "1D SE, 1D FD Extruded Domain horizontal divergence operator" begin
 
-    # Advection Equation
-    # ∂_t f + c ∂_x f  = 0
-    # the solution translates to the right at speed c,
-    # so if you you have a periodic domain of size [-π, π]
-    # at time t, the solution is f(x - c * t, y)
-    # here c == 1, integrate t == 2π or one full period
+    # Divergence Operator
+    # ∂_x (c f)
+    # for this test, we use f(x) = sin(x) and c = 1
+    # => ∂_x (c f) = cos(x)
 
-    function rhs!(dudt, u, _, t)
+    function rhs!(f)
```

Comment:
the Julia convention is that the `!` suffix should only be used when mutating arguments: https://docs.julialang.org/en/v1/manual/style-guide/#bang-convention

Since this no longer mutates the arguments, it should drop the `!` (and similar elsewhere)

Reply:
Agree. I am still not done cleaning up these tests. I will rename `rhs!` to something meaningful of the operator being tested in the test, as I have done in the following ones